socket.go

```diff
@@ -244,14 +252,45 @@ func (sck *socket) addConn(c *Conn) {
 	}
 	sck.ids[uuid] = c
 	if sck.r != nil {
-		sck.r.addConn(newMsgReader(c))
+		sck.r.addConn(c)
 	}
 	if sck.w != nil {
-		sck.w.addConn(newMsgWriter(c))
+		sck.w.addConn(c)
 	}
 	sck.mu.Unlock()
 }
 
+func (sck *socket) rmConn(c *Conn) {
+	sck.mu.Lock()
+	defer sck.mu.Unlock()
+
+	cur := 0
+	for i := range sck.conns {
+		if sck.conns[i] == c {
+			cur = i
+			break
+		}
+	}
+
+	if cur == -1 {
```

Comment:
that test doesn't look like it could possibly ever be true.
did you mean to initialize `cur` to `-1` at line 267 above?